### ISSUE TITLE
NodeShapes won't load images

### ISSUE THREAD

**Issue by ablikan** [dunnock/react-sigma#73]
I suspect it's because initPlugin() is not called in nodes.js, or if it is called, it's without the sigma instance, in any case, sigInst in drawImage always seems to be null.
I can't quite figure out how to call that function to feed it my instance, any hints?

The plugin otherwise works when it comes to supported shapes (square, cross...)

**Comment by dunnock** [comment]
Can you actually provide example?

**Comment by jaredkirkley** [comment]
Do you have an example of custom images working for nodes?  I'm having a tough time trying to get anywhere with that.

**Comment by uf0** [comment]
Hi @jaredkirkley,
do you have a working example?
I'm using your fork and I added a prop `image` to each node like stated here:
https://github.com/jacomyal/sigma.js/tree/master/plugins/sigma.renderers.customShapes#images

but no images appeared

my code looks like this:

```JSX
    <Sigma
      renderer="canvas"
      style={{ width: "100%", height: "100%" }}
      graph={{nodes:[{id:23, label: "20651609",size: 10,x: 501.10980224609375,y: -253.71058654785156,image:{url:"https://a0.muscache.com/im/pictures/a6aa21e9-d03c-4d16-8732-92b2bf891b5e.jpg"}]}}
    >
      <NodeShapes default="circle" />
    </Sigma>
```

**Comment by jaredkirkley** [comment]
I am using images that are local to my project which may be the difference.  My urls on my images are like `{ url: '/image.png' }`

I also wrote my own ImageNodes plugin component to include within the Sigma component.  This adds the images and type at render time though, so it may not be ideal for performance.

```
const ImageNodes = props => {
  useEffect(() => {
    if (props.sigma {
      props.sigma.graph.nodes().forEach(node => {
        node.image = '/image.png'
        node.type = 'circle'
      })
    }
  }, [props])

  return null
}

export default ImageNodes
```

**Comment by jaredkirkley** [comment]
@dunnock are you going to accept my PR for this fix?  I see you've worked on this project since this series of comments.  Please reply or make suggestions if you think my fix is not acceptable.

**Comment by dunnock** [comment]
Many thanks @jaredkirkley, I really appreciate that you took your time and found the root cause. Though I left comment in the pull request suggesting little change so that your fix will also work in cases when NodeShapes plugin is not loaded. Thanks again

**Comment by jaredkirkley** [comment]
@dunnock I'm not seeing any comments on the PR :(

Can you double check that the comment was posted?  I'd love to fix it up for you more!

Thanks for taking a look!

**Comment by dunnock** [comment]
My bad, I forgot to press submit review.. you should see it now

**Comment by dunnock** [comment]
@jaredkirkley  I have released your patch with 1.2.31, please confirm it works? Thanks

**Comment by dunnock** [comment]
Closing this as since resolution no comments
